### PR TITLE
New dashboard implementation

### LINKED ISSUE #806: JoinLock Custom Message
**Is your feature request related to a problem? Please describe.**
yes. current joinlock doesn't specify the reason for joinlock

**Describe the solution you'd like**
be able to set a custom text mesage as why the joinlock is switched on. so users who encounter joinlock, have some clue.

**Additional context**
moptop stolen situation after-lockdown

### PR DESCRIPTION
**Pull request by davipatury**
- Renames old `Modules` to `Controllers` (I'll completely remove all controllers soon)
- Implements new `Modules`
  - [x] PrefixModule
  - [x] LanguageModule
  - [x] JoinLockModule
- Closes #949 
- Closes #806 

###### [Web #45](https://github.com/SwitchbladeBot/bladey-web/pull/45)

### CODE REVIEW

**Comment by davipatury** [comment]
Currently no module has description, I am open to suggestions.

File: src/loaders/ControllerLoader.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   * Adds a new controller to the Client.
+   * @param {Controller} controller - Controller to be added
+   */
+  addController (controller) {
+    if (!(controller instanceof Controller)) {
+      this.log(`[31m${controller.name} failed to load - Not an Controller`, 'Controllers')
+      return false
+    }
+
+    if (controller.canLoad() !== true) {
```
Comment (metehus) [inline src/loaders/ControllerLoader.js]: why dont just `!controller.canLoad()`

File: src/modules/JoinLockModule.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    this.displayName = 'Join Lock'
+
+    this.defaultState = false
+    this.defaultValues = { message: '' }
+    this.specialInput = { message: { max: MAX_MESSAGE_SIZE } }
+  }
+
+  validateValues (entity) {
+    return Joi.validate(entity, Joi.object().keys({
+      message: Joi.string().max(MAX_MESSAGE_SIZE)
```
Comment (metehus) [inline src/modules/JoinLockModule.js]: maybe validate `.min()` and `.required()` too?

File: src/loaders/ControllerLoader.js (hunk 1 above)
Comment (davipatury) [reply src/loaders/ControllerLoader.js]: because `controller.canLoad()` can return an error message

File: src/modules/JoinLockModule.js (hunk 2 above)
Comment (davipatury) [reply src/modules/JoinLockModule.js]: it's not required, we have a default message

Comment (metehus) [reply src/modules/JoinLockModule.js]: oh yeah, and what ab `min(1)`? just for check

Comment (davipatury) [reply src/modules/JoinLockModule.js]: if the field is empty it fallbacks to the default message, so no problem

**Pull request merged**